src/scripts/choices.js

```diff
@@ -126,7 +126,18 @@ class Choices {
     this._wasTap = true;
     this._placeholderValue = this._generatePlaceholderValue();
     this._baseId = generateId(this.passedElement.element, 'choices-');
-    this._direction = this.passedElement.element.getAttribute('dir') || 'ltr';
+    /**
+     * setting direction in cases where it's explicitly set on passedElement
+     * or when calculated direction is different from the document
+     * @type {HTMLElement['dir']}
+     */
+    this._direction =
+      this.passedElement.element.dir ||
+      (({ direction }) =>
+        direction !==
+        window.getComputedStyle(document.documentElement).direction
+          ? direction
+          : '')(window.getComputedStyle(this.passedElement.element));
```

Comment:
This is not particularly easy to read - I think this could abstracted into it's own function or at least improved by using an if/else block

Reply:
C'mon, that lovely use of Javascript closure and ternary 😄 I think moving this functionality into an external one line function will make it even harder to understand, as you will need to check that function code. And this comparison have no use anywhere else in the library.
if/else you mean like this?

```js
this._direction = this.passedElement.element.dir;
if (!this._direction) {
  const { direction } = window.getComputedStyle(this.passedElement.element);
  if (window.getComputerStyle(document.documentElement).direction !== direction)
    this._direction = direction;
}
```
Do you prefer that version?

Reply:
This would be better:

```js
this._direction = this.passedElement.element.dir;

if (!this._direction) {
  const { direction: elementDirection } = window.getComputedStyle(this.passedElement.element);
  const { direction: documentDirection }  = window.getComputerStyle(document.documentElement);
  
  if (elementDirection !== documentDirection) {
    this._direction = direction; 
  }
}
```

If statements should always use brackets unless it's a ternary

Reply:
(i appreciate this lib is already inconsistent in that style but it's where i'd like to end up)

Reply:
actually i'm gunna open a pr to enforce this styling 👍